File: 792-binary-search/binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */

var binarySearch = function (nums, target, left, right) {
    const midIndex = left + Math.floor((right - left) / 2);
    if (nums[midIndex] === target) return midIndex;
    if(left >= right) return -1;
    if (nums[midIndex] > target) return binarySearch(nums, target, left, midIndex - 1);
    if (nums[midIndex] < target) return binarySearch(nums, target, midIndex + 1, right);
    // if(left === right )
}
var search = function (nums, target) {
    return binarySearch(nums, target, 0, nums.length-1);
    // Cheat way
    // return nums.indexOf(target);
};

File: 792-binary-search/binary-search.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search(nums, target)
    left_pointer = 0
    right_pointer = nums.length - 1
    while left_pointer <= right_pointer
        mid_pointer = (right_pointer + left_pointer)/2
        return mid_pointer if nums[mid_pointer] == target
        left_pointer = mid_pointer+1 if nums[mid_pointer] < target
        right_pointer = mid_pointer-1 if nums[mid_pointer] > target
    end
    -1
end